hoverxref/extension.py

```diff
@@ -232,21 +232,15 @@ def missing_reference(app, env, node, contnode):
         #   reftarget: float
         #   refexplicit: False
         inventories = InventoryAdapter(env)
-
-        # TODO: credits to https://github.com/readthedocs/sphinx-hoverxref/pull/144
-        # This chunk of code needs tests :)
-        reftype_fallbacks = {
-            'meth': 'method',
-            'mod': 'module',
-        }
+        env_domain = inventories.env.get_domain(domain)
+        # Intersphinx inventories may come with concrete types for domain roles.
+        # (e.g. for python, when reftype=='meth', inventory uses "method" or "classmethod"...
+        # so check in order, starting from the current reftype.
+        keys_to_check = [reftype, *env_domain.objtypes_for_role(reftype)]
```

Comment:
```suggestion
        keys_to_check = set([reftype, *env_domain.objtypes_for_role(reftype)])
```

In some cases `reftype` is already included in the response from `env_domain.objtypes_for_role(reftype)`, so we can create a `set()` to remove duplicated ones.